crates/dojo-world/src/manifest.rs

```diff
@@ -345,15 +345,19 @@ fn parse_contracts_events(
             let class_hash = data.next().expect("qed; missing class hash");
             let address = data.next().expect("qed; missing address");
 
-            upgrades
-                .entry(address)
-                .and_modify(|(current_block, current_class_hash)| {
-                    if *current_block < block_num {
-                        *current_block = block_num;
-                        *current_class_hash = class_hash;
-                    }
-                })
-                .or_insert((block_num, class_hash));
+            // Events that do not have a block number are ignored because we are unable to evaluate
+            // whether the events happened before or after the latest event that has been processed.
+            if let Some(num) = block_num {
+                upgrades
+                    .entry(address)
+                    .and_modify(|(current_block, current_class_hash)| {
+                        if *current_block < num {
+                            *current_block = num;
+                            *current_class_hash = class_hash;
+                        }
+                    })
+                    .or_insert((num, class_hash));
+            }
```

Comment:
block number in the events returned by the `getEvents` endpoint is [optional](https://github.com/starkware-libs/starknet-specs/blob/49665932a97f8fdef7ac5869755d2858c5e3a687/api/starknet_api_openrpc.json#L899-L932) in RPC 0.6.0